### PR TITLE
Sitemap Editor: Buttongrid support

### PR DESCRIPTION
**Pull request by mherwege**
This PR adds the Buttongrid element to the sitemap editor.
There is also some cleanup for configuration on small screens.
See https://github.com/openhab/openhab-core/pull/3810 and https://github.com/openhab/openhab-android/issues/3494

### CODE REVIEW

**Comment by mherwege** [comment]
@lolodomo I hope I have the JSON format right, but would need a working UI to test. I understand there is none yet. I will correct if there is an issue.

**Comment by lolodomo** [comment]
@mherwege : look at my message, I propose to change the syntax: https://github.com/openhab/openhab-core/issues/3441#issuecomment-1837590662
WDYT ?

**Comment by mherwege** [comment]
@lolodomo I am fine with that change. If it is agreed, I will adapt the sitemap editor PR accordingly.

**Comment by mherwege** [comment]
@lolodomo I now adapted it to the syntax with colons to separate row and column (`row:column:command`).

Here are few screenshots:

image

image

image

image

**Comment by florian-h05** [comment]
Is this finished and ready for review?

**Comment by mherwege** [comment]
> Is this finished and ready for review?

@florian-h05 Yes it is.

**Comment by lolodomo** [comment]
Is it compatible with the last syntax (PR merged yesterday) ?

**Review by ghys — APPROVED** [review]
Thanks @mherwege it seems to work well.

I think the header/tabs in the bottom pane (Widget, Visibility...) when on some phones with the iOS theme is becoming quite crowded, like this:

image

We might need to find a solution design-wise for these, when there are too many 'tabs' to display and there isn't enough room.

Otherwise LGTM!

File: bundles/org.openhab.ui/web/src/pages/settings/pages/sitemap/sitemap-edit.vue [hunk 1]
```diff
@@ -98,20 +105,32 @@
...
-    <f7-fab class="add-to-sitemap-fab" v-if="canAddChildren" position="right-bottom" slot="fixed" color="blue" @click="$refs.widgetTypeSelection.open()">
+    <f7-fab class="add-to-sitemap-fab" v-if="canAddChildren" position="right-center" slot="fixed" color="blue" @click="$refs.widgetTypeSelection.open()">
       <f7-icon ios="f7:plus" md="material:add" aurora="f7:plus" />
       <f7-icon ios="f7:multiply" md="material:close" aurora="f7:multiply" />
     </f7-fab>
 
-    <f7-sheet class="sitemap-details-sheet" :backdrop="false" :close-on-escape="true" :opened="detailsOpened" @sheet:closed="detailsOpened = false">
+    <f7-sheet v-if="currentTab === 'tree'" class="sitemap-details-sheet" :backdrop="false" :close-on-escape="true" :opened="detailsOpened" @sheet:closed="detailsOpened = false">
       <f7-page>
         <f7-toolbar tabbar bottom>
```
Comment (mherwege) [inline bundles/org.openhab.ui/web/src/pages/settings/pages/sitemap/sitemap-edit.vue:115]: Add ‘scrollable’ here may do.

**Comment by mherwege** [comment]
@ghys I added a scrollable attribute to the tabbar. I hope that makes it better.

**Comment by florian-h05** [comment]
I have tested this now on my iPhone 14 and in the Chrome DevTools with "Pixel 7" setting, and in both cases, the tab bar is now scrollable.
@ghys Please let me know if I can merge this or should wait for you.

**Comment by lolodomo** [comment]
The new widget is now implemented (merged) in Basic UI so we really need this PR before OH 4.1 code is frozen ;)

**Comment by florian-h05** [comment]
Sure. Let's wait for Yannick till Friday evening, and then merge,

**Review by ghys — APPROVED** [review]
I'm happy with this fix, maybe we should move the chevron (to collapse) to the left so we don't have to scroll to find it, but it'll do for now.
Thanks for addressing my concerns!

**Pull request merged**